src/server/worker/api_server.go

```diff
@@ -283,6 +292,7 @@ func NewAPIServer(pachClient *client.APIClient, etcdClient *etcd.Client, etcdPre
 		namespace:  namespace,
 		jobs:       ppsdb.Jobs(etcdClient, etcdPrefix),
 		pipelines:  ppsdb.Pipelines(etcdClient, etcdPrefix),
+		chunks:     col.NewCollection(etcdClient, path.Join(etcdPrefix, chunksPrefix), []col.Index{}, &Chunks{}, nil),
```

Comment:
Does `chunks` not need to be restricted to the job? How does the worker avoid claiming other jobs' datums?

Reply:
I think my names might be a bit confusing here. `chunks` is a collection that maps from `JobID` to a proto message defining what the chunks are. So there's only 1 value per job.

Then there's another collection which I call `locks` which is scoped on jobs and has a lock per chunk defined in the chunks message for the job.

Reply:
Also to clarify, I wound up doing something a bit simpler than what's in the design doc in that I have the master layout all the chunks at the beginning rather than workers dynamically laying out chunks as they go since the latter proved to be much trickier to get right. We could still implement the more complicated behavior if it feels like there's a limitation in this one, but I think we should wait until there's a good reason and just do the simple thing until then.